src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -168,86 +153,95 @@ private ModelTypeTable createTypeTable() {
     return apiMethods;
   }
 
-  /**
-   * Returns the Go package name.
-   *
-   * Retrieved by splitting the package string on `/` and returning the second to last string if
-   * possible, and otherwise the last string in the resulting array.
-   *
-   * TODO(saicheems): Figure out how to reliably get the intended value...
-   */
-  private static String getPackageName(String configPackageName) {
-    String split[] = configPackageName.split("/");
-    if (split.length - 2 >= 0) {
-      return split[split.length - 2];
+  private Collection<RetryPairDefinitionView> generateRetryPairDefinitions(
+      GoSurfaceNamer namer, SurfaceTransformerContext context) {
+    Set<RetryPairDefinitionView.Name> retryNames = new HashSet<>();
+    for (Method method : context.getNonStreamingMethods()) {
+      MethodConfig conf = context.getMethodConfig(method);
+      retryNames.add(
+          RetryPairDefinitionView.Name.create(
+              conf.getRetrySettingsConfigName(), conf.getRetryCodesConfigName()));
     }
-    return split[split.length - 1];
-  }
 
-  /**
-   * Returns the service's client name prefix.
-   */
-  private String getClientPrefix(Model model, Interface service, String packageName) {
-    String name = getReducedServiceName(service);
-    // If there's only one service, or the service name matches the package name, don't prefix with
-    // the service name.
-    if (model.getSymbolTable().getInterfaces().size() == 1 || name.equals(packageName)) {
-      return "";
+    TreeMap<RetryPairDefinitionView.Name, RetryPairDefinitionView> retryDef = new TreeMap<>();
+    Map<String, ImmutableSet<Code>> retryCodesDef =
+        context.getInterfaceConfig().getRetryCodesDefinition();
+    Map<String, RetrySettings> retryParamsDef =
+        context.getInterfaceConfig().getRetrySettingsDefinition();
+    for (RetryPairDefinitionView.Name name : retryNames) {
+      ImmutableSet<Code> codes = retryCodesDef.get(name.codes());
+      if (codes.isEmpty()) {
+        continue;
+      }
+      ImmutableSet.Builder<String> builder = ImmutableSet.<String>builder();
```

Comment:
don't name this for its structure (builder), name it for its use/meaning/intent (retryCodeSet) (+Builder if you want)